lib/remote_payload.coffee

```diff
@@ -0,0 +1,69 @@
+'use strict'
+
+#
+# CloudMine, Inc
+# 2016
+#
+
+_ = require 'lodash'
+
+DEFAULT_TIMEOUT_MS = 30000
+DEFAULT_BACKEND_VERSION = 1
+
+isTruthy = (value = '') ->
+  value.toString().toLowerCase() in ['true', 't', '1', 'yes']
+
+getReqPayload = (origReqPayload, contentType) ->
+  # json or form payloads will get merged into params
+  if contentType?.indexOf('application/json') is 0 or
+      contentType?.indexOf('application/x-www-form-urlencoded') is 0 or
+      contentType?.indexOf('multipart/form-data') is 0 or
+      contentType?.indexOf('text/plain') is 0
+    reqPayload = origReqPayload
+  reqPayload = {} unless reqPayload
+  reqPayload
+
+getExecutionParams = (reqQuery, reqPayload) ->
+  if reqQuery.f
+    execParams = {}
+    if reqQuery.params
+      try
+        execParams = JSON.parse(reqQuery.params)
+      catch e
+        execParams = reqQuery.params #as handled in platform
+  else
+    execParams = _.assign {}, (if _.isObject(reqPayload) then reqPayload else {}), reqQuery
+    delete execParams.apikey if execParams.apikey
```

Comment:
Yes, this just removes it from the "params" object if it was provided in the query string. See below, api_key gets populated under the "session" object